text/0000-roadmap-2017.md

```diff
@@ -0,0 +1,577 @@
+tooling, like the
+[compiler's error messages](https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html),
+that likewise have a big impact on productivity. There's so much more we can be
+doing in this space.
+
+In short, **productivity should be a core value of Rust**. By the end of 2017,
+let's try to earn the slogan:
+
+- Rust: fast, reliable, productive—pick three.
+
+# Detailed design
+[design]: #detailed-design
+
+## Overall strategy
+
+In the abstract, reaching the kind of adoption we need means bringing
+people along a series of distinct steps:
+
+- Public perception of Rust
+- First contact
+- Early play, toy projects
+- Public projects
+- Personal investment
+- Professional investment
+
+We need to (1) provide "drivers", i.e. strong motivation to continue through the
+stages and (2) avoid "blockers" that prevent people from progressing.
+
+At the moment, our most immediate adoption obstacles are mostly about blockers,
+rather than a lack of drivers: there are people who see potential value in Rust,
+but worry about issues like productivity, tooling, and maturity standing in the
+way of use at scale. The roadmap proposes a set of goals largely angled at
+reducing these blockers.
+
+However, for Rust to make sense to use in a significant way in production, it
+also needs to have a "complete story" for one or more domains of use. The goals
+call out a specific domain where we are already seeing promising production use,
+and where we have a relatively clear path toward a more complete story.
+
+Almost all of the goals focus squarely on "productivity" of one kind or another.
+
+## Goals
+
+Now to the meat of the roadmap: the goals. Each is phrased in terms of a
+*qualitative vision*, trying to carve out what the *experience* of Rust should
+be in one year's time. The details mention some possible avenues toward a
+solution, but this shouldn't be taken as prescriptive.
+
+> These goals are partly informed from the [internals thread] about the
+roadmap. That thread also posed a number of possible additional goals. Of
+course, part of the work of the roadmap is to allocate our limited resources,
+which fundamentally means not including some possible goals. Some of the most
+promising suggestions that didn't make it into the roadmap proposal itself are
+included at the end of the Goals section.
+
+### Rust should have a lower learning curve
+
+Rust offers a unique value proposition in part because it offers a unique
+feature: its ownership model. Because the concept is not (yet!) a widespread one
+in other languages, it is something most people have to learn from scratch
+before hitting their stride with Rust. And that often comes on top of other
+aspects of Rust that may be less familiar. A common refrain is "the first couple
+of weeks are tough, but it's oh so worth it." How many people are bouncing off
+of Rust before they get through those first couple of weeks? How many team leads
```

Comment:
That edit doesn't make sense to me, but I agree the original phrasing is a tad confusing. Perhaps "How many people give up on learning Rust during those first couple weeks?"